dags/multipod/configs/legacy_unit_test.py

```diff
@@ -0,0 +1,76 @@
+# Copyright 2024 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Utilities to run legacy tests from the old infrastructure."""
+
+import datetime
+import os
+from xlml.apis import gcp_config, metric_config, task, test_config
+from collections.abc import Iterable
+from dags import test_owner
+from dags.multipod.configs import common
+from dags.vm_resource import TpuVersion, Project, RuntimeVersion, ClusterName
+
+
+def get_legacy_unit_test_config(
+    test_cmd: Iterable[str],
+    tpu_version: TpuVersion,
+    tpu_cores: int,
+    tpu_zone: str,
+    time_out_in_min: int,
+    test_name: str,
+    test_mode: common.SetupMode,
+    test_owner: str,
+    docker_image: str,
+    num_slices: int = 1,
+    cluster_name: str = ClusterName.V4_8_MULTISLICE_CLUSTER.value,
+    project_name: str = Project.TPU_PROD_ENV_MULTIPOD.value,
+) -> task.TpuXpkTask:
+  """
+  Run a legacy unit test script.
+  test_cmd will be executed within the `dags/multipod/legacy_tests` directory.
+  """
+  job_gcp_config = gcp_config.GCPConfig(
+      project_name=project_name,
+      zone=tpu_zone,
+      dataset_name=metric_config.DatasetOption.XLML_DATASET,
+  )
+
+  run_model_cmds = (
```

Comment:
Should these be `set_up_cmds`?

Reply:
That's a good point. XPK doesn't support set_up_cmds at the moment, since there's no separation between setup and the actual test script, which is pretty confusing behavior. I'll fix this.

Reply:
Looks like the TestConfig does expose a setup script already: https://github.com/GoogleCloudPlatform/ml-auto-solutions/blob/b2f99ebfbad40e08a7a06134a9f050c53dfbed9d/xlml/apis/test_config.py#L245-L247

Reply:
XPK only accepts one workload command, and we're using the test script currently: https://github.com/GoogleCloudPlatform/ml-auto-solutions/blob/b2f99ebfbad40e08a7a06134a9f050c53dfbed9d/xlml/apis/task.py#L388

It would be nice to have a field which joins both setup and test commands.

Reply:
Yeah, I think original I only included run_model cmds as set_up cmds is inside of provision task group in GCE route and we don't have that here. I think we could make the refactor to make it reasonable in the future, since we already have diff routes, i.e. GKE via xpk, GKE via API call, GCE etc.

Reply:
How the setup command should be used is very context-specific. In GCE, it's either a metadata startup script or another SSH shell, but for GKE, the distinction isn't meaningful. In the multi-GPU test type, I just stuck them together (plus or minus some entrypoint hackery you can ignore) 

https://github.com/GoogleCloudPlatform/ml-auto-solutions/blob/d146d716c9eed97380afec3c2c894687faf1976f/xlml/apis/task.py#L674-L679

In general, we want to keep the "what" separate from the "how", so I think setup script is still useful, but the current structure of `Task` blurs the lines too much. We have some planned refactoring and can address this later.